internal/controller/progressive/progressive.go

```diff
@@ -471,7 +478,16 @@ func checkForUpgradeReplacement(
 	// If the new one is different from the existing Upgrading one:
 	//  Then if the new one matches the existing Promoted one: remove the Upgrading one
 	//  Else replace the Upgrading one with a new one
-	newUpgradingChildDef, err := makeUpgradingObjectDefinition(ctx, rolloutObject, controller, c, true)
+
+	existingUpgradingChildName := existingUpgradingChildDef.GetName()
+	// we need to create an Upgrading child definition which is template-evaluated using the existing Upgrading child's name so that we can effectively compare them below
+	newUpgradingChildDef, err := makeUpgradingObjectDefinition(ctx, rolloutObject, controller, c, &existingUpgradingChildName)
+	if err != nil {
+		return false, false, err
+	}
+	// we need to create an Upgrading child definition which is template-evaluated using the existing Promoted child's name so that we can effectively compare them below
+	existingPromotedChildName := existingPromotedChildDef.GetName()
+	newUpgradingChildDefUsingPromotedName, err := makeUpgradingObjectDefinition(ctx, rolloutObject, controller, c, &existingPromotedChildName)
```

Comment:
This is one of the core changes of the PR. When we compare the "promoted" child to the PipelineRollout, we must make sure that the PipelineRollout-defined spec and metadata are all evaluated using the "promoted" child's name so we can compare